volatility3/framework/layers/qemu.py

```diff
@@ -192,6 +263,45 @@ def _load_segments(self):
             else:
                 raise exceptions.LayerException(self._name, f'QEMU unknown section encountered: {section_byte}')
 
+    def _fallback_determine_architecture(self) -> str:
+        architecture_pattern = rb'pc-(i440fx|q35)-(\d{1,2}\.\d{1,2}|[\w\d\.]+)'
+        default_suffix = "-2.0"
+        base_layer = self.context.layers[self._base_layer]
+
+        vollog.log(constants.LOGLEVEL_VVVV, "QEVM fallback architecture detection used")
+
+        res = scanners.RegExScanner(architecture_pattern)
+        for offset in base_layer.scan(context = self.context, scanner = res):
+            line = base_layer.read(offset, 64)
+            regex_results = re.search(architecture_pattern, line)
+            architecture = regex_results.group().decode()
+            return architecture
+
+        # If that does not work, look in configuration JSON for devices specific to a certain architecture
+        architecture = None
+        for device in self._configuration.get('devices', []):
+            device_name = device.get('vmsd_name', '').lower()
+            if 'i440fx' in device_name or 'piix' in device_name:
+                architecture = 'pc-i440fx' + default_suffix
+                break
+            elif 'ich9' in device_name:
+                architecture = 'pc-q35' + default_suffix
+                break
+        if architecture:
+            return architecture
```

Comment:
We should maybe include a log message here that says we did establish the architecture, but that the version is just a guess. Currently we simply print "yeah, its version `2.0`", but we just don't know.